json_serializable/lib/src/utils.dart

```diff
@@ -162,6 +168,24 @@ ConstructorElement constructorByName(ClassElement classElement, String name) {
   return ctor;
 }
 
+Iterable<ClassElement> sealedClassImplementations(
+  ClassElement maybeSealedClass,
+) {
+  if (maybeSealedClass case final sc when sc.isSealed) {
+    return LibraryReader(sc.library)
+        .annotatedWith(
+          const TypeChecker.fromRuntime(JsonSerializable),
+        )
+        .map((e) => e.element)
+        .whereType<ClassElement>()
+        .where(
+          (e) => e.allSupertypes.contains(sc.thisType),
+        );
+  }
+
+  return [];
```

Comment:
If you want to be fancy 😄 

```suggestion
  return const Iterable<ClassElement>.empty();
```